component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteStore.java

```diff
@@ -0,0 +1,1158 @@
+     * updateOrAdd
+     */
+    @Override
+    protected CompiledCondition compileCondition(ExpressionBuilder expressionBuilder) {
+        ApacheIgniteConditionVisitor visitor = new ApacheIgniteConditionVisitor();
+        expressionBuilder.build(visitor);
+        return new ApacheIgniteCompiledCondition(visitor.returnCondition(), visitor.getParameters(),
+                visitor.getParametersConstant());
+    }
+
+    /**
+     * Compile the matching condition
+     *
+     * @param expressionBuilder that helps visiting the conditions in order to compile the condition
+     * @return compiled condition that can be used for matching events in find, contains, delete, update and
+     * updateOrAdd
+     */
+    @Override
+    protected CompiledExpression compileSetAttribute(ExpressionBuilder expressionBuilder) {
+        return compileCondition(expressionBuilder);
+    }
+
+    /**
+     * This method will be called before the processing method.
+     * Intention to establish connection to publish event.
+     *
+     * @throws ConnectionUnavailableException if end point is unavailable the ConnectionUnavailableException thrown
+     *                                        such that the  system will take care retrying for connection
+     */
+    @Override
+    protected void connect() throws ConnectionUnavailableException {
+        try {
+            Class.forName("org.apache.ignite.IgniteJdbcThinDriver");
+            connection = DriverManager.getConnection(connectionParams.toString());
+            this.createTable(primaryKey);
+            connected = true;
+        } catch (SQLException e) {
+            if (e.getSQLState().equals("08004")) {
+                throw new ApacheIgniteTableException("Either provided  username : " + username +
+                        " or password is incorrect for connecting to ignite store " + e.getMessage(), e);
+            } else {
+                throw new ConnectionUnavailableException("Failed to initialize Ignite store  " + e.getMessage(), e);
+            }
+        } catch (ClassNotFoundException e) {
+            throw new ApacheIgniteTableException("unable to find class IgniteJdbcThinDriver" + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Called after all publishing is done, or when {@link ConnectionUnavailableException} is thrown
+     * Implementation of this method should contain the steps needed to disconnect.
+     */
+    @Override
+    protected void disconnect() {
+        try {
+            if (connected) {
+                connection.close();
+            }
+        } catch (SQLException e) {
+            throw new ApacheIgniteTableException("unable to close the connection for ignite store.");
+        }
+    }
+
+    /**
+     * The method can be called when removing an event receiver.
+     * The cleanups that have to be done after removing the receiver could be done here.
+     */
+    @Override
+    protected void destroy() {
+
+        this.disconnect();
```

Comment:
Destroy is used to let go any resources you have kept, this will be called after disconnect. Do not call disconnect from destroy method.